src/main/scala/com/kinja/amqp/AmqpClient.scala

```diff
@@ -13,16 +13,37 @@ class AmqpClient(
 	val actorSystem: ActorSystem,
 	private val configuration: AmqpConfiguration,
 	private val logger: Slf4jLogger,
-	private val messageStore: MessageStore,
+	private val messageStores: Map[String, MessageStore],
```

Comment:
As a user of this library I'd ask what's the `String` here? It should be an enum at least. Why is this a `Map` and not a `List[(DeliveryGuarantee, MessageStore)]`, so essentially why can you only have one message store for a given type of guarantee? Please make such choices better explained by using better designed types. `Map` and `String` are just too generic.

Reply:
I understand your objection against `String`, but `Map`? One message store for one type of guarantee is exactly what we need.

Reply:
This was exactly the kind of decision that was not obvious for me. Maybe some documentation could explain this.

Reply:
I've added a few words about that to the README.